packages/thycotic_ss/data_stream/logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,495 @@
+        "SECRET - SECRETPOLICYCHANGE":                  { "category": ["iam"], "type": ["change"] }
+        "SECRETTEMPLATE - CREATE":                      { "category": ["iam"], "type": ["creation"] }
+        "SECRETTEMPLATE - EDIT":                        { "category": ["iam"], "type": ["change"] }
+        "SECRETTEMPLATE - TEMPLATE COPIED FROM":        { "category": ["iam"], "type": ["creation"] }
+        "SECRETTEMPLATE - FIELD ENCRYPTED":             { "category": ["iam"], "type": ["info"] }
+        "SECRETTEMPLATE - FIELD EXPOSED":               { "category": ["iam"], "type": ["info"] }
+        "SECRETPOLICY - CREATE":                        { "category": ["iam"], "type": ["creation"] }
+        "SECRETPOLICY - EDIT":                          { "category": ["iam"], "type": ["change"] }
+        "EXPORTSECRETS - EXPORTED":                     { "category": ["iam"], "type": ["access"] }
+        "IMPORTSECRETS - IMPORTED":                     { "category": ["iam"], "type": ["change"] }
+        "UNLIMITEDADMIN - ENABLE":                      { "category": ["iam"], "type": ["change"] }
+        "UNLIMITEDADMIN - DISABLE":                     { "category": ["iam"], "type": ["change"] }
+        "USERAUDIT - EXPIRENOW":                        { "category": ["iam"], "type": ["info"] }
+        "LICENSES - EXPIRES30DAYS":                     { "category": ["configuration"], "type": ["info"] }
+        "POWERSHELLSCRIPT - CREATE":                    { "category": ["iam"], "type": ["creation"] }
+        "POWERSHELLSCRIPT - DEACTIVATE":                { "category": ["iam"], "type": ["change"] }
+        "POWERSHELLSCRIPT - EDIT":                      { "category": ["iam"], "type": ["change"] }
+        "POWERSHELLSCRIPT - REACTIVATE":                { "category": ["iam"], "type": ["change"] }
+        "POWERSHELLSCRIPT - VIEW":                      { "category": ["iam"], "type": ["access"] }
+      source: >
+        def schemaId = ctx.cef.name.toString();
+        def schema = params[schemaId];
+        if (schema != null) {
+          if (ctx.event == null) {
+            ctx.event = new HashMap();
+          }
+          ctx.event.category = schema.category;
+          ctx.event.type = schema.type;
+        }
+
+  - grok:
+      field: cef.name
+      patterns:
+        - '%{DATA:event.provider} - %{GREEDYDATA:event.action}'
+        - '%{DATA:event.provider} %{GREEDYDATA:event.action}'
+        - '%{GREEDYDATA:event.action}'
+      ignore_missing: true
+      description: "Parse cef.name into more usable ECS event.provider and event.action fields"
+
+  - lowercase:
+      field: event.provider
+      ignore_missing: true
+
+  - lowercase:
+      field: event.action
+      ignore_missing: true
+
+  - convert:
+      field: cef.device.event_class_id
+      type: string
+      target_field: event.code
+      ignore_missing: true
+      description: "The CEF name is the basic event description used by Thycotic SS"
```

These converts with type string can be reduced to sets since the type of the values are already string.

```suggestion
  - set:
      field: event.code
      copy_from: cef.device.event_class_id
      description: "The CEF name is the basic event description used by Thycotic SS"
```
Similar below.

---

```
Device Event Class ID
This is a unique identifier per event-type. This can be a string or an integer. The Device Event Class ID identifies the type of event reported. In the Intrusion Detection System (IDS) world, each signature or rule that detects certain activity has a unique Device Event Class ID assigned.
```

If we use set we run the risk of putting an integer into event.code, when event.code is supposed to be a keyword/string. I will admit I have no checked the code behind filebeat's `decode_cef` parser, so perhaps it will always output a string?

---

OK. Thanks.